### DISCUSSION TITLE
Is this a bug or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by danielwlewis** [ImageMagick/ImageMagick#7083] [Help] [answered]
### ImageMagick version

 7.1.1-21

### Operating system, version and so on

Windows 10

### Description

When I try the command,

`magick img.jpg -format '%[EXIF:*]' -identify -`

with the attached image file (img.jpg), it displays a number of EXIF parameters, but followed by a lot of what looks like must be binary data. How can I prevent displaying the binary data, or is this a bug?

Thanks!
Dan

![img](https://github.com/ImageMagick/ImageMagick/assets/42775951/1af32db2-4809-41ad-a949-5a717da1020d)

**Comment by snibgo** [comment]
You are writing the (binary, JPG format) image to "-". If you want just the text EXIF metadata, without the image, then write to `info:`.
```
magick danImg.jpg -format %[EXIF:*] -identify info:
```

**Comment by fmw42** [comment]
I do not think your syntax is correct.  Perhaps the command you want is either

`magick img.jpg -format '%[EXIF:*]' info:`

or

`magick identify -format '%[EXIF:*]' img.jpg`

**Reply by danielwlewis** [reply]
Hi snibgo and Fred,

Thanks, but I still seem to have a problem. Although your suggested solutions work fine from the command line, (I think that) sending the output to info: is somehow incompatible with the way I'm using IM. I hope you can suggest a workaround.

The attached ZIP file ([main.zip](https://github.com/ImageMagick/ImageMagick/files/14143863/main.zip)) contains the C program I'm using as a test. It basically loads and executes magick from another program using the Windows CreateProcess function and creates pipes to magick's stdin and stdout streams so that I can send commands to magick and read its responses using a separate thread. Specifically, I am trying to do two things for each of about 16,000 jpeg files - (1) optionally create a thumbnail and (2) optionally extract the EXIF metadata. I give the following command line to CreateProcess:

`magick -define jpeg:size=160x200 -script -`

Then send magick four lines of input similar to the following for each jpeg. (The second line is omitted when a thumbnail is not required, and the third is omitted when the EXIF metadata is not required).

```
img.jpg
-thumbnail 80x100 -write thumb.jpg
-format '%[EXIF:*]' -identify info:
-delete 0
```

Finally I send one last input string after all the jpgs:

`-exit`

What happens is that after displaying the EXIF data for the first image, it outputs the following and then prematurely exits:

     magick: no decode delegate for this image format `INFO' @ error/constitute.c/ReadImage/746.

Any suggestions?

Thanks again!
Dan

**Comment by snibgo** [comment, marked as answer]
```
img.jpg
-thumbnail 80x100 -write thumb.jpg
-format '%[EXIF:*]' -identify info:
-delete 0
```
That part is wrong. When writing to a file or a pseudo-format such as `info:` or `txt:`, the full option is `-write` followed by the filename. However, if that is the final option in a command line, the `-write` keyword is omitted. If we want to write *within* the command, we need the `-write` keyword. If you omit the keyword, IM thinks you want to *read* from `info:`, which can't be done, hence the error. So you need:
```
img.jpg
-thumbnail 80x100 -write thumb.jpg
-format '%[EXIF:*]' -identify -write info:
-delete 0
```

**Reply by danielwlewis** [reply]
Thank you! That seems to have corrected the problem. I never would have gotten it without your help!

Best,
Dan

**Reply by danielwlewis** [reply]
One last question: Is there a way to exclude MakerNotes from the returned EXIF metadata?

Best,
Dan

**Reply by snibgo** [reply]
No. Your format can include EXIF tags like `%[EXIF:ApertureValue]` or all of then with `%[EXIF:*]`, but there is no way to specify "all of them except MakerNotes". See https://imagemagick.org/script/escape.php

**Reply by fmw42** [reply]
You could perhaps pipe to sed and filter that tag out.
